packages/mantine/src/components/table/layouts/row-layout/RowLayoutIcon.tsx

```diff
@@ -1,16 +1,13 @@
 import {Icon as IconType, ListSize16Px} from '@coveord/plasma-react-icons';
-import {CompoundStylesApiProps, factory, Factory, useProps} from '@mantine/core';
-import {useRowLayout} from './RowLayoutContext';
+import {CompoundStylesApiProps, Factory, factory, useProps} from '@mantine/core';
 
-export type RowLayoutIconStylesNames = 'icon';
```

Comment:
The way its setup right now we can't, because when we switch to another layout, the RowLayoutContext is not wrapping the table anymore so it throws an error when trying to render the layout segmented control. I was thinking that allowing to override the icon through the default props was enough.